src/vs/editor/contrib/goToDeclaration/goToDeclaration.ts

```diff
@@ -20,38 +20,49 @@ function getDefinitions<T>(
 	model: ITextModel,
 	position: Position,
 	registry: LanguageFeatureRegistry<T>,
-	provide: (provider: T, model: ITextModel, position: Position, token: CancellationToken) => Location | Location[] | Thenable<Location | Location[]>
-): TPromise<Location[]> {
+	provide: (provider: T, model: ITextModel, position: Position, token: CancellationToken) => Location | Location[] | SymbolDefinition | Thenable<Location | Location[] | SymbolDefinition>
+): TPromise<SymbolDefinition> {
```

Comment:
This changes how the corresponding API-command works. We should update the doc around that and inform live share folks